pkg/resource/deploy/step_generator.go

```diff
@@ -72,8 +72,51 @@ func (sg *stepGenerator) isTargetedUpdate() bool {
 	return sg.updateTargetsOpt != nil || sg.replaceTargetsOpt != nil
 }
 
-func (sg *stepGenerator) isTargetedForUpdate(urn resource.URN) bool {
-	return sg.updateTargetsOpt == nil || sg.updateTargetsOpt[urn]
+// isTargetedForUpdate returns if urn is target for update. The function accommodates
+// `--target-dependents`. `targetDependentsForUpdate` should probably be called if this function
+// returns true.
+func (sg *stepGenerator) isTargetedForUpdate(
```

Comment:
Offhand it looks like these changes always use the `--target-dependents` behavior--I would have expected us to terminate early if `sg.opts.TargetDependents` is false.